src/jit/importer.cpp

```diff
@@ -1086,8 +1085,17 @@ GenTreePtr Compiler::impAssignStruct(GenTreePtr           dest,
     }
 
     assert(dest->gtOper == GT_LCL_VAR || dest->gtOper == GT_RETURN || dest->gtOper == GT_FIELD ||
-           dest->gtOper == GT_IND || dest->gtOper == GT_OBJ);
+           dest->gtOper == GT_IND || dest->gtOper == GT_OBJ || dest->gtOper == GT_INDEX);
 
+    if (dest->OperGet() == GT_LCL_VAR && src->OperGet() == GT_LCL_VAR &&
+        src->gtLclVarCommon.gtLclNum == dest->gtLclVarCommon.gtLclNum)
+    {
+        // Make this a NOP
+        return gtNewNothingNode();
+    }
+
+    // TODO-1stClassStructs: Avoid creating an address if it is not needed,
+    // or re-creating a Blk node if it is.
```

Comment:
Not sure I understood this comment. In what cases we can avoid creating an addr/blk node that we are not avoiding?  This comments needs to be expanded with those details or if it not relevant should be deleted.